### PR TITLE
Ass2/conf parsing

### PR DESCRIPTION
**Pull request by samueleceroni**

### CODE REVIEW

File: Assignment2/superserver.c [hunk 1]
```diff
@@ -29,18 +38,121 @@ typedef struct {
...
+ServiceDataVector read_configuration(FILE *stream) {
+	// Initialize the vector
+	ServiceDataVector config;
+	long fileBegin = ftell(stream);
+	config.size = count_lines(stream);
+	fseek(stream, fileBegin, SEEK_SET);
+	config.services = (ServiceData*)malloc(config.size * sizeof(ServiceData));
+
+	// Generate the format string to read parameters from a line
+	char formatString[7*4];
```
Comment (samueleceroni) [inline Assignment2/superserver.c:95]: 7*4? ahahah

File: Assignment2/superserver.c [hunk 2]
```diff
@@ -29,18 +38,121 @@ typedef struct {
...
+
+	// Read and parse line by line
+	char line[MAX_LINE_SIZE];
+	for (size_t index = 0; index < config.size; index++) {
+		do {
+			char* result = fgets(line, MAX_LINE_SIZE, stream);
+			if (result == NULL && !feof(stream)) {
+				die(EXIT_READ_ERROR);
+			}
+		} while(is_empty(line));
```
Comment (samueleceroni) [inline Assignment2/superserver.c:107]: would do in a separate function, but that's fine

File: Assignment2/superserver.c [hunk 3]
```diff
@@ -29,18 +38,121 @@ typedef struct {
...
+
+		ServiceData *current = &config.services[index];
+
+		int count = sscanf(line, formatString,
+			current->path, current->protocol, current->port, current->mode);
+		if (count != 4 ||
+			(strcmp(PROTOCOL_UDP, current->protocol) != 0 && strcmp(PROTOCOL_TCP, current->protocol) != 0) ||
+			(strcmp(MODE_WAIT, current->mode) != 0 && strcmp(MODE_NOWAIT, current->mode) != 0) ||
+			(atoi(current->port) <= 0 || atoi(current->port) > PORT_MAX)) {
+			die(EXIT_CONF_ERROR);
```
Comment (samueleceroni) [inline Assignment2/superserver.c:117]: I wuold check the parameter in another function but it's fine

**Review by samueleceroni — COMMENTED** [review]
Looks fine to me, forgive (or ignore) my obsession against function longer than 10 lines

File: Assignment2/superserver.c (hunk 1 above)
Comment (Carlovan) [reply Assignment2/superserver.c:95]: 4 parameters, 7 characters for each

**Pull request closed without merging**